pkg/discovery/metrics/metric.go

```diff
@@ -154,7 +154,7 @@ type Metric interface {
 	GetMetricProp() MetricProp
 	GetUID() string
 	GetValue() interface{}
-	UpdateValue(existing interface{})
+	UpdateValue(existing interface{}, maxMetricPointsSize int) Metric
```

Comment:
Suggestion: Consider setting `maxMetricPointsSize` while creating the metrics as part of `NewEntityResourceMetric()`, rather then in each `UpdateValue()` call.

Reply:
Discussed with Irfan, will keep current logic which will be cleaner and less code changes.